include/deal.II/lac/tensor_product_matrix.h

```diff
@@ -1164,6 +1327,136 @@ TensorProductMatrixSymmetricSumCollection<dim, Number, n_rows_1d>::finalize()
 
       for (const auto &i : cache)
         store(i.second, i.first);
+
+      if (true)
+        {
+          if (dim == 1)
+            {
+              this->inverted_eigenvalues.resize(this->eigenvalues.size());
+              for (unsigned int i = 0; i < this->eigenvalues.size(); ++i)
+                this->inverted_eigenvalues[i] =
+                  Number(1.0) / this->eigenvalues[i];
```

Comment:
Also here, there is a lot of duplicated code. I think we should break this out into a function `fill_inverted_eigenvalues` or similarly.